tests/lib/utils/common.js

```diff
@@ -90,6 +98,32 @@ function testRule(
       code: `<div style={${ruleCode}} />`,
       options: [{ nativeAllowList: ["style"] }],
     },
+    {
+      code: `<Item style={${ruleCode}} />`,
+      options: [{ ignoreComponents: ["Item"] }],
+    },
+    {
+      code: `import {Item as ItemFoo} from "react-foo";<ItemFoo style={${ruleCode}} />`,
+      options: [{ ignoreSources: ["react-foo"] }],
+    },
+    {
+      code: `import {Item as ItemFoo} from "react-foo";<ItemFoo style={${ruleCode}} />`,
+      options: [
+        { ignoreSources: [{ source: "react-foo", importNames: ["Item"] }] },
+      ],
+    },
+    {
+      code: `import Foo from "react-foo";<Foo.Item style={${ruleCode}} />`,
+      options: [
+        { ignoreSources: [{ source: "react-foo", importNames: ["Item"] }] },
```

Comment:
would it make sense to instead namespace as `importNames: ["Foo.Item"]`? wdyt?

Reply:
Do you mean to drop the `source` property?

Reply:
I think we need to adjust the config to be able to, for example, allow the known patterns for `Foo.Item`, but disallow for `Bar.Item`. 

```
import Foo from "react-foo"
import Bar from "react-bar"

render() {
  return <><Foo.Item /><Bar.Item /></>
}

```

The options in the PR on line 118 don't distinguish between `Foo` and `Bar`.

Reply:
Hmm, I didn't get it yet. It does disallow for `Bar.Item` since `Bar` is imported from `react-bar`.